src/operator/tensor/elemwise_unary_op_basic.cc

```diff
@@ -350,10 +350,108 @@ NNVM_REGISTER_OP(_identity_with_attr_like_rhs)
 .add_argument("lhs", "NDArray-or-Symbol", "First input.")
 .add_argument("rhs", "NDArray-or-Symbol", "Second input.");
 
+void ReshapeLikeRangeCanonicalize(int ndims, const char *side, int begin,
+                                  const dmlc::optional<int> &end, int *cbegin,
+                                  int *cend) {
+  *cbegin = begin;
+  if (*cbegin < 0)
+    *cbegin += ndims;
+
+  if (!static_cast<bool>(end)) {
```

Comment:
you can use has_value for better readability.

Reply:
Ok, that is cleaner. Fixed.

Btw: the reason I used this casting method because it was done [here](https://github.com/apache/incubator-mxnet/blob/51f650e0bf3b905fec4aebfc873c1c56eac61536/src/operator/tensor/matrix_op-inl.h#L1125). Should this be changed as well?